src/common/offline_transformations/src/compress_quantize_weigths.cpp

```diff
@@ -211,86 +224,630 @@ ov::pass::CompressQuantizeWeights::CompressQuantizeWeights() {
+            *zero_point++ = zero_point_value;
+            return scale;
         });
-        if (!is_transformed_and_original_equal)
-            return false;
+}
 
-        std::shared_ptr<Node> new_weights = std::make_shared<opset8::Subtract>(weights, int8_zero_point);
-        OPENVINO_SUPPRESS_DEPRECATED_START
-        if (auto constant = ov::get_constant_from_source(new_weights)) {
-            OPENVINO_SUPPRESS_DEPRECATED_END
-            new_weights = constant;
-        } else {
-            return false;
+bool compute_scale_and_zero_point(const std::shared_ptr<ov::op::v0::Constant>& output_low,
+                                  const std::shared_ptr<ov::op::v0::Constant>& output_high,
+                                  size_t levels,
+                                  ov::Tensor& scale_tensor,
+                                  ov::Tensor& zero_point_tensor,
+                                  bool& zero_point_is_zero) {
+    const auto type = output_low->get_element_type();
+    switch (type) {
+    case ov::element::Type_t::f32: {
+        compute_scale_and_zero_point_internal<float>(output_low,
+                                                     output_high,
+                                                     levels,
+                                                     scale_tensor,
+                                                     zero_point_tensor,
+                                                     zero_point_is_zero);
+        break;
+    }
+    case ov::element::f16: {
+        compute_scale_and_zero_point_internal<ov::float16>(output_low,
+                                                           output_high,
+                                                           levels,
+                                                           scale_tensor,
+                                                           zero_point_tensor,
+                                                           zero_point_is_zero);
+        break;
+    }
+    default:
+        return false;
+    }
+
+    return true;
+}
+
+template <typename T, typename U, typename F>
+static void
+transform(const T* first1, const T* const last1, const T* first2, const T* first3, const T* first4, U* out, F& f) {
+    while (first1 < last1) {
+        *out++ = f(*first1++, *first2++, *first3++, *first4++);
+    }
+}
+
+template <typename T, typename U, typename F>
+static void transform(const T* first1,
+                      const T* const last1,
+                      const T* first2,
+                      const T* first3,
+                      const T* first4,
+                      const T* first5,
+                      const T* first6,
+                      U* out,
+                      F& f) {
+    while (first1 < last1) {
+        *out++ = f(*first1++, *first2++, *first3++, *first4++, *first5++, *first6++);
+    }
+}
+
+template <typename T, typename U, typename F>
+static void numpy_broadcast_4inputs(const T* weights,
```

Comment:
Maybe can we combine these functions and the functions implemented in fq reference and move them to some utils?

Reply:
Maybe when we have the code that actually needs it outside of this transformation. I would prefer not to expose these symbols prematurely.